### PR TITLE
Adding changes for FADA longetivity test automation

### PR DESCRIPTION
**Pull request by jainsmit**
**What this PR does / why we need it**:
Adding code for FA Direct Access Volume automation
1. Adding a code to support both CSI and Portworx volume plugin
2. Adding a code to check if volume is pure volume or not
3. Not performing operation on pure volumes which are not supported
4. Adding a code to set the Topology in k8s node
5. Adding a code to set Topology info in PX nodes
6. Adding a code to validate Topology
7. Adding a code for csi snapshots and validation
8. Adding a code for csi snap restore and validation

**Which issue(s) this PR fixes** (optional)
Closes # PTX-6731

**Special notes for your reviewer**:

1. Completed the dev run successfully for portworx provisioner:
https://jenkins.pwx.dev.purestorage.com/job/Dev/job/torpedo-dev03/31/

2.Ran the CSI job successfully using my torpedo image
https://jenkins.pwx.dev.purestorage.com/job/Torpedo/view/Torpedo-Detailed/job/tp-k8s-matrix-op-csi/250/parameters/

3. Tested the code with Daemonset base installation
http://jen-endurance.pwx.dev.purestorage.com/job/Torpedo/job/tp-fada-longevity/39/console

### CODE REVIEW

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -1204,8 +1225,13 @@ func (k *K8s) createStorageObject(spec interface{}, ns *corev1.Namespace, app *s
...
 
-		logrus.Infof("Setting provisioner of %v to %v", obj.Name, volume.GetStorageProvisioner())
-		obj.Provisioner = volume.GetStorageProvisioner()
+		if volume.GetStorageProvisioner() != PortworxStrict {
+			logrus.Infof("Setting provisioner of %v to %v", obj.Name, volume.GetStorageProvisioner())
+			obj.Provisioner = volume.GetStorageProvisioner()
+		} else {
+			logrus.Infof("Setting provisioner of %v to %v", obj.Name, obj.Provisioner)
+		}
+
```
Comment (mborodin-px) [inline drivers/scheduler/k8s/k8s.go:1234]: you can remove else here as it is redundant
```
if volume.GetStorageProvisioner() != PortworxStrict {
			obj.Provisioner = volume.GetStorageProvisioner()
} 
logrus.Infof("Setting provisioner of %v to %v", obj.Name, obj.Provisioner)
```

File: drivers/scheduler/scheduler.go [hunk 2]
```diff
@@ -323,6 +329,27 @@ type Driver interface {
 
 	//RecyleNode deletes nodes with given node
 	RecycleNode(n node.Node) error
+
+	// DeletePXPods delete PX pods
+	DeletePXPods(nameSpace string) error
```
Comment (mborodin-px) [inline drivers/scheduler/scheduler.go:334]: we already have DeleteTask for that, please use it, no need Portworx specific method

**Review by lsrinivas-pure — COMMENTED** [review]
LGTM. please address the Maksym comments

File: drivers/scheduler/scheduler.go (hunk 2 above)
Comment (jainsmit) [reply drivers/scheduler/scheduler.go:334]: @mborodin-px I tried using DeleteTask but how to use it for portworx pods controller by StorageCluster..

File: drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (jainsmit) [reply drivers/scheduler/k8s/k8s.go:1234]: Done..

File: drivers/scheduler/k8s/specs/elasticsearch/pure/pure-storage-class.yaml [hunk 3]
```diff
@@ -12,8 +12,5 @@ parameters:
   max_iops: "30000"
   max_bandwidth: "10G"
   csi.storage.k8s.io/fstype: ext4
-  createoptions: -b 4096
-mountOptions:
-  - nosuid
-  - discard
-allowVolumeExpansion: true
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
```
Comment (Pure-AdamuKaapan) [inline drivers/scheduler/k8s/specs/elasticsearch/pure/pure-storage-class.yaml:16]: [nit] Add newline at end of file

Comment (jainsmit) [reply drivers/scheduler/k8s/specs/elasticsearch/pure/pure-storage-class.yaml]: Done !!

**Review by nikolaypopov — APPROVED** [review]

**Comment by jainsmit** [comment]
Another Dev run after incorporating the review comments:
https://jenkins.pwx.dev.purestorage.com/job/Dev/job/torpedo-dev03/32/

**Pull request merged**